autocomplete/complete.bash

```diff
@@ -4,38 +4,40 @@
 # Copyright (C) 2016 Arvid Gerstmann
 #
 
+# usage: _tldr_get_files [architecture] [semi-completed word to search]
 _tldr_get_files() {
-	local ret
-	local files="$(find $HOME/.tldrc/tldr/pages/$1 -name '*.md' -exec basename {} .md \;)"
-
-	IFS=$'\n\t'
-	for f in $files; do
-	    echo $f
-	done
+    find "$HOME"/.tldrc/tldr/pages/"$1" -name "$2"'*.md' -exec basename -s .md {} +
```

Comment:
I would recommend the reviewers to test if this is better, or using `find` to only search for `'*.md'` instead of `"$2"'*.md'`.

Comparing the performance of `_tldr_complete` function using `time` command does not help, since both methods perform almost the same.

Using the following commands, I was able to determine that after applying this change, the bottleneck is the `find` command:
```
$ word="he"
$ cmpl=`find $HOME/.tldrc/tldr/pages/linux/ -name "$word"'*.md' -exec basename -s '.md' {} + && find $HOME/.tldrc/tldr/pages/common/ -name "$word"'*.md' -exec basename -s '.md' {} +`; cmpl_sorted_n_uniq=`printf "%s" "$cmpl" | sort | uniq`
$ compgen -W "$cmpl_sorted_n_uniq" -- "$word" | pv -lr > /dev/null
$ ( find $HOME/.tldrc/tldr/pages/linux/ -name "$word"'*.md' -exec basename -s '.md' {} + && find $HOME/.tldrc/tldr/pages/common/ -name "$word"'*.md' -exec basename -s '.md' {} + ) | pv -rl >/dev/null
```
The results will be in this form:
```
[compgen's speed]
[find's speed]
```

In the other case (using only `-name '*.md'` in find command), I found `compgen` was the bottleneck instead of `find`; using the following commands:
```
$ word="he"
$ cmpl=`find $HOME/.tldrc/tldr/pages/linux/ -name '*.md' -exec basename -s '.md' {} + && find $HOME/.tldrc/tldr/pages/common/ -name '*.md' -exec basename -s '.md' {} +`; cmpl_sorted_n_uniq=`printf "%s" "$cmpl" | sort | uniq`
$ compgen -W "$cmpl_sorted_n_uniq" -- "$word" | pv -lr > /dev/null
$ ( find $HOME/.tldrc/tldr/pages/linux/ -name '*.md' -exec basename -s '.md' {} + && find $HOME/.tldrc/tldr/pages/common/ -name '*.md' -exec basename -s '.md' {} + ) | pv -rl >/dev/null
```
The result will be in the same form as described above.

---

The results I got:

1. with `find ... -name '*.md' ...`: 
    - `compgen`'s speed: 1.69k/s
    - `find`'s speed: 241k/s
2. with `find ... -name "$2"'*.md' ...`: 
    - `compgen`'s speed: 255k/s
    - `find`'s speed: 1.48k/s

Both methods used about 0.05 secs on my system (using `time _tldr_complete`).

Reply:
In theory `find.... | xargs -P "$(nproc)" basename...` hour potentially be faster. I wonder if using a pure-bash alternative to basename or an awk/sed solution there could help? The trouble is that you're spawning a new process for each line with the basename call there.